docs/reference/crud.txt

```diff
@@ -11,6 +11,188 @@ CRUD Operations
    :class: singlecol
 
 
+Accessing Field Values
+======================
+
+Mongoid provides several ways of accessing field values.
+
+.. note::
+
+  All of the access methods described below raise
+  ``ActiveModel::MissingAttributeError`` when the field being accessed is
+  :ref:`projected out <projection>`, either by virtue of not being included in
+  :ref:`only <only>` or by virtue of being included in
+  :ref:`without <without>`. This applies to both reads and writes.
+
+
+Getters & Setters
+-----------------
+
+The recommended way is to use the getter and setter methods generated for
+each declared field:
+
+.. code-block:: ruby
+
+  class Person
+    include Mongoid::Document
+
+    field :first_name
+  end
+
+  person = Person.new
+
+  person.first_name = "Artem"
+  person.first_name
+  # => "Artem"
+
+To use this mechanism, each field must be explicitly declared, or the
+model class must enable :ref:`dynamic fields <dynamic-fields>`.
+
+
+Custom Getters & Setters
+------------------------
+
+It is possible to explicitly define the getter and setter methods to provide
+custom behavior when reading or writing fields, for example value
+transformations or storing values under different field names. In this case
+``read_attribute`` and ``write_attribute`` methods can be used to read and
+write the values directly into the attributes hash:
+
+.. code-block:: ruby
+
+  class Person
+    include Mongoid::Document
+
+    def first_name
+      read_attribute(:fn)
```

Comment:
why is this read_attribute(:fn) I would think the purpose of using read_attribute is when you override the getter of the same name? 

So if you had:
```
class Person 
  include Mongoid::Document
  field :first_name, type: String

  def first_name
    fn = read_attribute(:first_name)
    # some modifications to fn
  end    
end
```
This way you could get the value for :first_name without calling self.first_name and causing an infinite loop.

Reply:
Otherwise, if the field is called fn, just call `self.fn`

Reply:
This example shows how to implement storage field names (which an application wouldn't do because this functionality is provided by Mongoid itself). The read is intentionally of another attribute than what the model exposes.